ebos/eclwriter.hh

```diff
@@ -361,16 +414,23 @@ private:
                 if (c1 > c2)
                     continue; // we only need to handle each connection once, thank you.
 
-
+                // Ordering of compressed and uncompressed index should be the same
+                assert(cartesianCellIdx[c1] <= cartesianCellIdx[c2]);
                 int gc1 = std::min(cartesianCellIdx[c1], cartesianCellIdx[c2]);
                 int gc2 = std::max(cartesianCellIdx[c1], cartesianCellIdx[c2]);
-                if (gc2 - gc1 == 1)
+
+                if (gc2 - gc1 == 1) {
                     tranx.data[gc1] = globalTrans->transmissibility(c1, c2);
+                    continue; // skip other if clauses as they are false, last one needs some computation
+                }
 
-                if (gc2 - gc1 == cartDims[0])
+                if (gc2 - gc1 == cartDims[0]) {
                     trany.data[gc1] = globalTrans->transmissibility(c1, c2);
+                    continue; // skipt next if clause as it needs some computation
+                }
 
-                if (gc2 - gc1 == cartDims[0]*cartDims[1])
+                if ( gc2 - gc1 == cartDims[0]*cartDims[1] ||
+                     directVerticalNeighbors(Dune::CartesianIndexMapper<Grid>(globalGrid_), cartesianToActive, gc1, gc2))
```

Comment:
There's one possible optimisation here.  You're forming a new `CartesianIndexMapper`, but the implementation of the `directVerticalNeighbors` function "only" requires the global Cartesian dimensions.  Could you not pass a reference to the [`cartDims`](https://github.com/OPM/opm-simulators/blob/4d31f116051b39ed4f8e9bd0b3ea9974b71668f5/ebos/eclwriter.hh#L364) object that already exists instead of a full mapper object?

Reply:
Indeed, Nice catch. I amended the commit accordingly.